Easydict/Swift/Service/OpenAI/LLMStreamService.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  LLMStreamService.swift
+//  Easydict
+//
+//  Created by tisfeng on 2024/5/20.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - LLMStreamService
+
+@objcMembers
+@objc(EZLLMStreamService)
+public class LLMStreamService: QueryService {
+    // MARK: Public
+
+    override public func isStream() -> Bool {
+        true
+    }
+
+    override public func intelligentQueryTextType() -> EZQueryTextType {
+        Configuration.shared.intelligentQueryTextTypeForServiceType(serviceType())
+    }
+
+    override public func supportLanguagesDictionary() -> MMOrderedDictionary<AnyObject, AnyObject> {
+        let allLangauges = EZLanguageManager.shared().allLanguages
+        let supportedLanguages = allLangauges.filter { language in
+            !unsupportedLanguages.contains(language)
+        }
+
+        let orderedDict = MMOrderedDictionary<AnyObject, AnyObject>()
+        for language in supportedLanguages {
+            orderedDict.setObject(language.rawValue as NSString, forKey: language.rawValue as NSString)
+        }
+        return orderedDict
+    }
+
+    override public func queryTextType() -> EZQueryTextType {
+        var typeOptions: EZQueryTextType = []
+
+        let isTranslationEnabled = UserDefaults.bool(forKey: EZTranslationKey, serviceType: serviceType())
+        let isSentenceEnabled = UserDefaults.bool(forKey: EZSentenceKey, serviceType: serviceType())
+        let isDictionaryEnabled = UserDefaults.bool(forKey: EZDictionaryKey, serviceType: serviceType())
+
+        if isTranslationEnabled {
+            typeOptions.insert(.translation)
+        }
+        if isSentenceEnabled {
+            typeOptions.insert(.sentence)
+        }
+        if isDictionaryEnabled {
+            typeOptions.insert(.dictionary)
+        }
+
+        return typeOptions
+    }
+
+    override public func serviceUsageStatus() -> EZServiceUsageStatus {
+        let usageStatus = UserDefaults.string(forKey: EZServiceUsageStatusKey, serviceType: serviceType()) ?? ""
+        guard let value = UInt(usageStatus) else { return .default }
+        return EZServiceUsageStatus(rawValue: value) ?? .default
+    }
+
+    // MARK: Internal
+
+    let throttler = Throttler()
+    var updateCompletion: ((EZQueryResult, Error?) -> ())?
+
+    var model = ""
+
+    var unsupportedLanguages: [Language] {
+        []
```

Comment:
Could we change these default return values to `fatalError("Not implemented")`, that will remind us to override these in subclasses.

Reply:
I think it's fine to keep it that way, for most LLM services we don't need to specify unsupported languages for several reasons:

- LLM services usually don't explicitly tell us all the languages they support, unless we test them all, which is too much trouble.
- LLM services usually keep updating their versions, and the languages they support may change as well, making it difficult for us to keep up with it.
- Most importantly, LLM services are not very sensitive to language types, and sometimes it can get correct translation results even if it recognizes the wrong language type.

For example, here is a Latin sentence, which is incorrectly recognized as Dutch because we don't support that language in Easydict at the moment, but even so, most LLM services translate it correctly, which is great 😃

### [Mundus vult decipi, ergo decipiatur.](https://weibo.com/6827625527/Of2Rm0k0s)


<img width="418" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/0a5846b4-3eaf-427b-895f-f1082b1a2a85">

### [Google Translate](https://translate.google.com/?sl=auto&tl=zh-CN&text=Mundus%20vult%20decipi%2C%20ergo%20decipiatur.&op=translate)
<img width="1342" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/2989791e-6655-48eb-b74c-56ca2b07db43">

Reply:
Ok, that sounds fine. What about the other functions like  `availableModels ` / `apiKey ` / 'endpoint', in case of we forget to implement them in subclass, maybe consider using `fatal()`.

Reply:
Good idea, I have marked `model`,`availableModels`, `apiKey` and `endpoint` as must be overriden.